dns/ddclient/src/opnsense/mvc/app/models/OPNsense/DynDNS/Migrations/M1_2_0.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+/*
+ * Copyright (C) 2021 Deciso B.V.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ * INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+ * AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+ * OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+namespace OPNsense\DynDNS\Migrations;
+
+use OPNsense\Base\BaseModelMigration;
+use OPNsense\Base\FieldTypes\BooleanField;
+use OPNsense\Base\FieldTypes\NetworkField;
+use OPNsense\Core\Config;
+
+class M1_2_0 extends BaseModelMigration
+{
+    /**
+     * Migrate older models into shared model
+     * @param $model
+     */
+    public function run($model)
+    {
+        $config = Config::getInstance()->object();
+
+        if (empty($config->OPNsense->DynDNS)) {
+            return;
+        }
+
+        $globalsettings = $config->OPNsense->DynDNS->general;
+        $accounts = $config->OPNsense->DynDNS->accounts;
+
+        foreach ($accounts as $account) {
```

Comment:
To iterate the model, you should use the $model parameter, for example:

https://github.com/opnsense/core/blob/ef6f5a268cd9589f54f5af911c716d9e1b5752d7/src/opnsense/mvc/app/models/OPNsense/Cron/Migrations/M1_0_2.php#L40-L41

The old settings can only be retrieved from the config (as you're doing now), updating the config object isn't safe as the model will be flushed later as well (overwriting the config settings).

Reply:
Just fixed it with [7b39e52](https://github.com/opnsense/plugins/pull/2883/commits/7b39e529dbbe4f27814b36b5860f315b6ffb310a)

Thanks for the input. It's my first ever plugin edit here ;)